src/common/lib/python/common_extension.c

```diff
@@ -317,6 +319,20 @@ static int PyTask_init(PyTask *self, PyObject *args, PyObject *kwds) {
     }
   }
   utarray_free(val_repr_ptrs);
+  /* Set the resource vector of the task. */
+  if (resource_vector != NULL) {
+    CHECK(PyList_Size(resource_vector) == MAX_RESOURCE_INDEX);
+    for (int i = 0; i < MAX_RESOURCE_INDEX; ++i) {
+      PyObject *resource_entry = PyList_GetItem(resource_vector, i);
+      task_spec_add_required_resource(self->spec, i,
+                                      PyFloat_AsDouble(resource_entry));
+    }
+  } else {
+    for (int i = 0; i < MAX_RESOURCE_INDEX; ++i) {
+      task_spec_add_required_resource(self->spec, i,
+                                      i == CPU_RESOURCE_INDEX ? 1.0 : 0.0);
```

Comment:
This code isn't very generalizable to other resources. I would either leave a comment explaining that these are the default values for these particular indices, or make a `static double[]` containing the default values.